cli/src/main/java/org/jboss/pnc/bacon/cli/App.java

```diff
@@ -17,156 +17,190 @@
-import org.aesh.command.Executor;
-import org.aesh.command.GroupCommandDefinition;
-import org.aesh.command.activator.CommandActivator;
-import org.aesh.command.activator.OptionActivator;
-import org.aesh.command.builder.CommandBuilder;
-import org.aesh.command.completer.CompleterInvocation;
-import org.aesh.command.converter.ConverterInvocation;
-import org.aesh.command.impl.registry.AeshCommandRegistryBuilder;
-import org.aesh.command.invocation.CommandInvocation;
-import org.aesh.command.parser.CommandLineParserException;
-import org.aesh.command.parser.OptionParserException;
-import org.aesh.command.parser.RequiredOptionException;
-import org.aesh.command.registry.CommandRegistry;
-import org.aesh.command.settings.SettingsBuilder;
-import org.aesh.command.validator.CommandValidatorException;
-import org.aesh.command.validator.ValidatorInvocation;
-import org.aesh.readline.Prompt;
-import org.aesh.readline.ReadlineConsole;
-import org.aesh.readline.terminal.formatting.Color;
-import org.aesh.readline.terminal.formatting.TerminalColor;
-import org.aesh.readline.terminal.formatting.TerminalString;
+import org.fusesource.jansi.AnsiConsole;
 import org.jboss.bacon.da.Da;
-import org.jboss.pnc.bacon.common.cli.AbstractCommand;
+import org.jboss.pnc.bacon.common.Constant;
+import org.jboss.pnc.bacon.common.ObjectHelper;
+import org.jboss.pnc.bacon.common.cli.VersionProvider;
 import org.jboss.pnc.bacon.common.exception.FatalException;
+import org.jboss.pnc.bacon.config.Config;
 import org.jboss.pnc.bacon.pig.Pig;
 import org.jboss.pnc.bacon.pnc.Pnc;
+import org.jline.builtins.Builtins;
+import org.jline.builtins.SystemRegistry;
+import org.jline.builtins.SystemRegistryImpl;
+import org.jline.builtins.Widgets;
+import org.jline.keymap.KeyMap;
+import org.jline.reader.Binding;
+import org.jline.reader.EndOfFileException;
+import org.jline.reader.LineReader;
+import org.jline.reader.LineReaderBuilder;
+import org.jline.reader.MaskingCallback;
+import org.jline.reader.Parser;
+import org.jline.reader.Reference;
+import org.jline.reader.UserInterruptException;
+import org.jline.reader.impl.DefaultParser;
+import org.jline.terminal.Terminal;
+import org.jline.terminal.TerminalBuilder;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+import picocli.shell.jline3.PicocliCommands;
 
-import java.util.Arrays;
-import java.util.stream.Collectors;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static picocli.CommandLine.ScopeType.INHERIT;
 
 /**
  * @author Michal Szynkiewicz, michal.l.szynkiewicz@gmail.com <br>
  *         Date: 12/13/18
  */
 
 @Slf4j
-@GroupCommandDefinition(name = "bacon", description = "Bacon CLI", groupCommands = { Pnc.class, Da.class, Pig.class })
-public class App extends AbstractCommand {
+@Command(name = "bacon.jar", versionProvider = VersionProvider.class, subcommands = { Da.class, Pig.class, Pnc.class })
```

Comment:
why .jar in the name?

Reply:
Er, no reason :-) It does mean I can remove `commandLine.setCommandName("bacon");`. Will push a change.

Reply:
I guess the "official" way to run it is via the `bacon` script as opposed to `java -jar bacon.jar`, therefore, I think bacon for the same makes sense.